website/static/js/accountSettings.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+var $ = require('jquery');
+var $osf = require('js/osfHelpers');
+var ko = require('knockout');
+var oop = require('js/oop');
+var Raven = require('raven-js');
+
+require('knockout.punches');
+ko.punches.enableAll();
+
+
+var UserEmail = oop.defclass({
+    constructor: function() {
+        this.address = ko.observable();
+        this.isConfirmed = ko.observable();
+        this.isPrimary = ko.observable();
+    }
+});
+
+
+var UserProfile = oop.defclass({
+    constructor: function () {
+
+        this.id = ko.observable();
+        this.emails = ko.observableArray();
+
+        this.primaryEmail = ko.computed(function () {
+            var emails = this.emails();
+            for (var i = 0; i < this.emails().length; i++) {
+                if(emails[i].isPrimary()) {
+                    return emails[i];
+                }
+            }
+            return new UserEmail();
+        }.bind(this));
+
+        this.alternateEmails = ko.computed(function () {
+            var emails = this.emails();
+            var retval = [];
+            for (var i = 0; i < this.emails().length; i++) {
+                if (emails[i].isConfirmed() && !emails[i].isPrimary()) {
+                    retval.push(emails[i]);
+                }
+            }
+            return retval;
+        }.bind(this));
+
+        this.unconfirmedEmails = ko.computed(function () {
+            var emails = this.emails();
+            var retval = [];
+            for (var i = 0; i < this.emails().length; i++) {
+                if(!emails[i].isConfirmed()) {
+                    retval.push(emails[i]);
+                }
+            }
+            return retval;
+        }.bind(this));
+
+    }
+});
+
+
+var UserProfileClient = oop.defclass({
+    constructor: function () {
+        this.urls = {
+            fetch: '/api/v1/profile/',
+            update: '/api/v1/profile/'
+        };
+    },
+    fetch: function () {
+        var ret = $.Deferred();
+
+        var request = $.get(this.urls.fetch);
+        request.done(function(data) {
+            var profile = new UserProfile();
+            this.unserialize(data, profile);
+            ret.resolve(profile);
+
+        }.bind(this));
+        request.fail(function(xhr, status, error) {
+            $osf.growl('Error', 'Could not fetch user profile.', 'danger');
+            Raven.captureMessage('Error fetching user profile', {
+                url: this.urls.fetch,
+                status: status,
+                error: error
+            });
+        }.bind(this));
+
+        return ret.promise();
+    },
+    update: function (profile) {
+        var request = $osf.putJSON(
+            this.urls.update,
+            this.serialize(profile)
+        ).done(function (data) {
+            profile = this.unserialize(data, profile);
+        }.bind(this)).fail(function(xhr, status, error) {
```

Comment:
I find it more clear and concise to do `var self = this;` rather than having to bind everywhere.